Tools/CMake/AMReXParallelBackends.cmake

```diff
@@ -143,6 +143,12 @@ if (  AMReX_GPU_BACKEND STREQUAL "CUDA"
       endif ()
    endif ()
 
+   # Flags to make it an error to write a device variable in
+   # a host function.
+   if (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 11.2)
+      list(APPEND _cuda_flag --display-error-number --diag-error 20092)
```

Comment:
Great idea!
So far we make a similarish warning (capture `this`) a warning by default and only error it in CI with an extra option. The basic idea behind this is that "shipping" any kind of "-Werror" is usually considered bad practice and hurts adoption in unexpected environments (e.g. with other/new/in-development compilers).